js/views/userPage/UserPage.js

```diff
@@ -0,0 +1,113 @@
+import $ from 'jquery';
+import BaseVw from '../baseVw';
+import loadTemplate from '../../utils/loadTemplate';
+import app from '../../app';
+import Home from './UserPageHome';
+import Store from './UserPageStore';
+import Follow from './UserPageFollow';
+
+export default class extends BaseVw {
+  constructor(options = {}) {
+    super(options);
+    this.options = options;
+
+    this.tabViewCache = {};
+    this.tabViews = { Home, Store, Follow };
+
+    this.followed = false; // TODO check to see if user is followed by the viewer
+    this.followsYou = true; // TODO check to see if this user follows the viewer
+    this.ownPage = true; // TODO check to see if this is the viewer's page
+  }
+
+  className() {
+    return 'userPage';
+  }
+
+  events() {
+    return {
+      'click .js-tab': 'tabClick',
+      'click .js-followBtn': 'followClick',
+      'click .js-messageBtn': 'messageClick',
+      'click .js-moreBtn': 'moreClick',
+    };
+  }
+
+  tabClick(e) {
+    const targ = $(e.target).closest('.js-tab');
+    this.selectTab(targ);
+  }
+
+  followClick() {
+    // TODO add in follow functionality
+    if (this.followed) {
+      // unfollow this user
+      console.log('unfollow');
+      // do the following as the callback of the unfollow action
+      this.followed = false;
+      this.$followLbl.removeClass('hide');
+      this.$unfollowLbl.addClass('hide');
+    } else {
+      // do the following as the callback of the follow action
+      console.log('follow');
+      this.followed = true;
+      this.$followLbl.addClass('hide');
+      this.$unfollowLbl.removeClass('hide');
+    }
+  }
+
+  messageClick() {
+    // activate the chat message
+    console.log('message');
+  }
+
+  moreClick() {
+    this.$moreableBtns.toggleClass('hide');
+  }
+
+  selectTab(targ) {
+    let tabViewName = targ.data('tab');
+    let tabView = this.tabViewCache[tabViewName];
+    const tabViewType = tabViewName; // the original view name is passed in for the Follow view
+
+    this.$tabTitle.text(tabViewName);
+
+    if (tabViewName === 'Followers' || tabViewName === 'Following') tabViewName = 'Follow';
+
+    if (!this.currentTabView || this.currentTabView !== tabView) {
+      this.$('.js-tab').removeClass('clrT active');
+      targ.addClass('clrT active');
+      if (this.currentTabView) this.currentTabView.$el.detach();
+      if (!tabView) {
+        tabView = new this.tabViews[tabViewName]({ tabViewType, ownPage: this.ownPage });
+        this.tabViewCache[tabViewName] = tabView;
+        tabView.render();
+      }
+      this.$tabContent.append(tabView.$el);
+      this.currentTabView = tabView;
+    }
+  }
+
+  render() {
+    loadTemplate('userPage/userPage.html', (t) => {
+      this.$el.html(t({
+        ...app.profile.toJSON(),
```

Comment:
I don't think that we're going to want to be sending app.profile in here, but rather this.model, which might be the same as app.profile, but only if we're on our own page.